### PR TITLE
Laboratory work #8, Anastasia Skachkova - 20FPL2

### PR DESCRIPTION
**Pull request by awakenedlilac**

### CODE REVIEW

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix CI tests

File: lab_7_llm/main.py [hunk 1]
```diff
@@ -6,9 +6,13 @@
 from pathlib import Path
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Checkout the files from 7 lab to the master

File: lab_8_llm/main.py [hunk 2]
```diff
@@ -150,6 +208,27 @@ def infer_sample(self, sample: tuple[str, ...]) -> str | None:
...
+            return None
+
+        tokens = self._tokenizer(sample[0],
+                                 sample[1],
+                                 max_length=120,
+                                 padding=True,
+                                 return_tensors='pt',
+                                 truncation=True)
+
+        with torch.no_grad():
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py]: why ` torch.no_grad()` used?

File: lab_8_llm/start.py [hunk 3]
```diff
@@ -2,15 +2,26 @@
...
     """
-    result = None
+    settings = LabSettings(PROJECT_ROOT / "lab_8_llm" / 'settings.json')
+
+    ds_obtainer = RawDataImporter(settings.parameters.dataset)
+    ds_obtainer.obtain()
+
+    result = RawDataPreprocessor(ds_obtainer.raw_data)
+    print(result.analyze())
+    # result = None
```
Comment (SalnikovIgor) [inline lab_8_llm/start.py]: Unskip or delete

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+datasets==2.16.1
```
Comment (SalnikovIgor) [inline requirements.txt:2]: Add line `--extra-index-url https://download.pytorch.org/whl/cpu`

File: seminars/seminar_01_16_2024/assets/danetqa_example.csv [hunk 5]
```diff
@@ -6,8 +6,96 @@ question,passage,idx,label
 Был ли кетчуп в читосе?,Текущий каталог продукции размещен на сайте производителя. Кетчуп  Колбаска и бекон  Краб  Луковые кольца  Паприка  Пицца  Рождественская индейка  Сметана и лук  Стейк на пылающем гриле  Сыр и лук  Сырный сыр  Original Копчёный бекон  Барбекю стейк  Соль и уксус  Паприка на гриле  Соус табаско  Оригинальный  Сыр по-гречески  Тайские специи  Красный перец  Итальянский песто  Лёгкая сметана и лук  Стейк-гриль и карамелизованный лук  Морская соль и бальзамический уксус  Тёртый созревший сыр и зелёный лук  Тайский сладкий чили и цитронелла  Соль и уксус  Горячий и острый  Красная паприка  Тайский чили и лайм  Сыр и лук  Сметана и лук  Sweet BBQ Spare Rib Сыр и лук  Техасский барбекю-соус  Оригинальный  Сметана и лук  Соль и уксус  Blastin' Buffalo Wing Вопиющий укропный рассол  Взрывные сыр и чили  Огненный соус чили  Огненный васаби  Экстра сырный начо сыр  Экстра сметана и лук  Экстра дерзкий барбекю  Экстра пряный соус чили,4,1
```
Comment (SalnikovIgor) [inline seminars/seminar_01_16_2024/assets/danetqa_example.csv]: Delete from PR

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_8_llm/main.py (hunk 2 above)
Comment (awakenedlilac) [reply lab_8_llm/main.py]: i forgot to remove this line...)

File: lab_8_llm/start.py (hunk 3 above)
Comment (awakenedlilac) [reply lab_8_llm/start.py]: done

File: requirements.txt (hunk 4 above)
Comment (awakenedlilac) [reply requirements.txt:2]: done

File: seminars/seminar_01_16_2024/assets/danetqa_example.csv (hunk 5 above)
Comment (awakenedlilac) [reply seminars/seminar_01_16_2024/assets/danetqa_example.csv]: done

File: seminars/seminar_01_16_2024/assets/danetqa_example.csv [hunk 6]
```diff
@@ -1,13 +0,0 @@
-question,passage,idx,label
```
Comment (SalnikovIgor) [inline seminars/seminar_01_16_2024/assets/danetqa_example.csv:1]: Delete from PR

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix all comments

File: seminars/seminar_01_16_2024/assets/danetqa_example.csv (hunk 6 above)
Comment (awakenedlilac) [reply seminars/seminar_01_16_2024/assets/danetqa_example.csv:1]: i have nothing in this file locally :(

File: lab_7_llm/main.py (hunk 1 above)
Comment (awakenedlilac) [reply lab_7_llm/main.py]: done

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Files lab 7 and dataset are still in the PR

**Pull request closed without merging**